cupyx/jit/_builtin_funcs.py

```diff
@@ -219,6 +219,46 @@ def call_const(self, env, dtype, size, alignment=None):
         env.locals[name] = var
         return Data(name, _cuda_types.Ptr(ctype))
 
+class LocalMemory(BuiltinFunc):
+
+    def __call__(self, dtype, size, alignment=None):
+        """Allocates local memory and returns it as a 1-D array.
+
+        Args:
+            dtype (dtype):
+                The dtype of the returned array.
+            size (int or tuple):
+                If ``int`` type, the size of static local memory.
+                If ``tuple`` type, the size of a multi-dimensional static local memory.
+                Does not use __extern__ keyword like shared memory does
+            alignment (int or None): Enforce the alignment via __align__(N).
+        """
+        super().__call__()
+        
+    def call(self, env: 'Environment', *args, **kwargs):
+        if(type(args[1]) is Data): #intercept tuple, convert to Constant(tuple)
+            args = list(args)
+            _int_list = args[1].code.split("(")[1].strip(")").split(", ")
+            if(len(_int_list) == 1): #convert 1-element tuple to integer (1d array)
+                args[1] = Constant(int(_int_list[0]))
+            else:
+                for i in range(len(_int_list)):
+                    _int_list[i] = Constant(int(_int_list[i])).obj
+                args[1] = Constant(tuple(_int_list))
+            args = tuple(args)
+        return super().call(env, *args, **kwargs)
```

Comment:
```py
@jit.rawkernel()
def f():
    n = 10
    lmem = jit.local_memory(numpy.int32, n)

f[1, 1]()
```

The above code raises `IndexError: list index out of range`. If you intend to support only constant size, you may remove these lines.